client/src/pages/private/SiteTable.js

```diff
@@ -23,14 +43,99 @@ const columns = [
   { id: 'details' },
 ];
 
-export default () => {
+const SiteFormsDialog = ({ activeForm, onDialogSubmit, onDialogClose }) => {
   const { openToast } = useToast();
+
+  const handleSiteCreate = async (site) => {
+    const response = await fetch(`${API_URL}/api/v1/employer-sites`, {
+      method: 'POST',
+      headers: {
+        Authorization: `Bearer ${store.get('TOKEN')}`,
+        Accept: 'application/json',
+        'Content-type': 'application/json',
+      },
+      body: JSON.stringify(site),
+    });
+
+    if (response.ok) {
+      onDialogClose();
+      await onDialogSubmit();
+    } else {
+      const error = await response.json();
+      if (error.status && error.status === 'Duplicate') {
+        openToast({ status: ToastStatus.Error, message: 'Duplicate site ID' });
+      } else {
+        openToast({
+          status: ToastStatus.Error,
+          message: response.error || response.statusText || 'Server error',
+        });
+      }
+    }
+  };
+
+  return (
+    <Dialog title='Create Site' open={activeForm != null} onClose={onDialogClose}>
+      {activeForm === 'new-site' && (
+        <NewSiteForm
+          initialValues={{
+            siteId: '',
+            siteName: '',
+            registeredBusinessName: '',
+            address: '',
+            city: '',
+            isRHO: null,
+            postalCode: '',
+            healthAuthority: '',
+            allocation: '',
+            operatorName: '',
+            operatorContactFirstName: '',
+            operatorContactLastName: '',
+            operatorPhone: '',
+            operatorEmail: '',
+            siteContactFirstName: '',
+            siteContactLastName: '',
+            siteContactPhone: '',
+            siteContactEmail: '',
+          }}
+          validationSchema={CreateSiteSchema}
+          onSubmit={(values) => {
+            handleSiteCreate({
```

Comment:
```suggestion
            handleSiteCreate({...values, siteId: parseInt(values.siteId), allocation: parseInt(values.allocation)})
```
Is this mapping logic required?

Reply:
I didn't change anything here ☝️
I just moved it to a new component

Reply:
I am suggesting, just check that mapping logic has any merit or not